internal/cluster/connection_manager.go

```diff
@@ -192,6 +192,9 @@ func (m *ConnectionManager) start(ctx context.Context) error {
 	} else if addr, err = m.startUnisocket(ctx); err != nil {
 		return err
 	}
+	if err = m.sendStateToCluster(ctx); err != nil {
```

Comment:
Yes, `ConnectionManager.start` is executed whenever the client reconnects.

Reply:
Also, where do we have the logic of not allowing invocations until the state is sent?

Reply:
> What happens when this fails? In Java, we retry sending the state again. Also, in Java, there are other parts of the state that might apply to the Go client, like local proxies. I believe we should also send them (maybe can be done in a separate PR, but I believe we should do it)

If sending state to the cluster fails during `StartNewClient`, ConnectionManager start fails too, and then `StartNewClient` fails. This may happen if the client disconnects from the cluster right before sending the state.

I've updated the code to retry at: https://github.com/hazelcast/hazelcast-go-client/pull/910/commits/45ba429f57eb87c4c2a80dd10d6aa393ece3a2a2

I found a bug during testing this, and created this ticket: https://github.com/hazelcast/hazelcast-go-client/issues/934

Reply:
> where do we have the logic of not allowing invocations until the state is sent?

I've added some logic to pause sending non-urgent invocations at: https://github.com/hazelcast/hazelcast-go-client/pull/910/commits/7471a22a203060a34d4436fa8a174cd8ade08e13

https://github.com/yuce/hazelcast-go-client/blob/ad4231ffbac8ec70c57a8c235e732b44b415cb0c/internal/invocation/invocation_service.go#L121